File: talkadvisor-domain/src/main/java/org/craftsrecords/talkadvisor/recommendation/api/CreateProfile.java
package org.craftsrecords.talkadvisor.recommendation.api;

import org.craftsrecords.talkadvisor.recommendation.preferences.Preferences;
import org.craftsrecords.talkadvisor.recommendation.profile.Profile;

import javax.annotation.Nonnull;

@FunctionalInterface
public interface CreateProfile {
    @Nonnull
    Profile forUserWithPreferences(@Nonnull String userId, @Nonnull Preferences preferences);
}

File: talkadvisor-domain/src/main/kotlin/org/craftsrecords/talkadvisor/recommendation/api/CreateProfile.kt
package org.craftsrecords.talkadvisor.recommendation.api

import org.craftsrecords.talkadvisor.recommendation.preferences.Preferences
import org.craftsrecords.talkadvisor.recommendation.profile.Profile

@FunctionalInterface
interface CreateProfile {
    fun forUserWithPreferences(userId: String, preferences: Preferences): Profile
}
